src/gui/rss/articlelistwidget.cpp

```diff
@@ -55,6 +55,7 @@ void ArticleListWidget::setRSSItem(RSS::Item *rssItem, bool unreadOnly)
 {
     // Clear the list first
     clear();
+    m_rssArticleToListItemMapping.clear();
```

Comment:
I'm hoping for it. You might want to consider adding 
```c++
Q_ASSERT(count() == m_rssArticleToListItemMapping.count());
```
at the end of this function.